client/my-sites/plans/current-plan/jetpack-checklist/index.tsx

```diff
@@ -85,6 +85,36 @@ class JetpackChecklist extends PureComponent {
 		return <ChecklistSectionTitle excludeFromCount title={ title } />;
 	}
 
+	renderTaskSet( checklistTasks ) {
+		return map( this.props.taskStatuses, ( status, taskId ) => {
+			const task = checklistTasks[ taskId ];
+
+			if ( ! task ) {
+				// UI does not support this task.
+				return;
+			}
```

Comment:
seems odd to me to return `undefined` inside the `map`. I'm guessing `React` handles it well but did you consider filtering out the unwanted results first?

```ts
const allTasks = map( this.props.taskStatuses, ( status, taskId ) => [
	checklistTasks[ taskId ], 
	get( status, [ 'completed' ], false ),
	taskId
] );
const tasks = filter( allTasks, ( [ task ] ) => !! task );

return map( tasks, ( [ task, completed, taskId ] ) => (
	<Task
		completed={ completed }
		…
	/>
) );
```

also, if we're using `map()` and `filter()` from `lodash` because these things may not be available and may no be arrays, we could consider defaulting to an empty array. it's not clear to me if this could be an object…

```ts
( this.props.taskStatuses || [] )
	.map( ( status, taskId ) => [ checklistTasks[ taskId ], !! status.completed, taskId ] )
	.fitler( ( [ task ] ) => !! task )
	.map( ( [ task, completed, taskId ] ) => ( … ) );
```

Reply:
I agree, this is strange, but also handled gracefully by React. I kept this largely as it was but I'll revisit it.